packages/react/src/components/TreeView/__tests__/TreeNode-test.js

```diff
@@ -8,125 +8,119 @@
-        ref={mockRef}
-      />
-    );
-
+    renderWithProviders(<TreeNode id="node1" label="Node 1" ref={mockRef} />);
     expect(mockRef.current).not.toBeNull();
     expect(mockRef.current?.tagName).toBe('LI');
-    expect(container.querySelector('li')).not.toBeNull();
-    expect(container.querySelector('li')?.textContent).toBe('Node 1');
   });
 
-  it('should pass the correct props to child TreeNodes', () => {
-    const { getByText } = render(
-      <TreeNode
-        id="parent"
-        label="Parent Node"
-        depth={1}
-        active="child1"
-        disabled={false}
-        selected={['child1']}
-        onTreeSelect={() => {}}>
-        <TreeNode id="child1" label="Child Node 1" disabled={false} />
+  // UPDATED: This test now uses renderWithProviders to supply context
+  it('should apply correct attributes based on context', () => {
+    const providerProps = {
+      active: 'child1',
+      selected: ['child1'],
+    };
+    const { getByText } = renderWithProviders(
+      <TreeNode id="parent" label="Parent Node">
+        <TreeNode id="child1" label="Child Node 1" />
         <TreeNode id="child2" label="Child Node 2" disabled={true} />
-      </TreeNode>
+      </TreeNode>,
+      { providerProps, depth: 0 }
     );
 
     const child1 = getByText('Child Node 1').closest('li');
     const child2 = getByText('Child Node 2').closest('li');
 
     expect(child1).toHaveAttribute('id', 'child1');
-    expect(child1).toHaveAttribute('aria-selected', 'true');
+    expect(child1).toHaveAttribute('aria-selected', 'true'); // Verified via context
     expect(child1).toHaveAttribute('tabindex', '-1');
 
     expect(child2).toHaveAttribute('id', 'child2');
     expect(child2).toHaveAttribute('aria-disabled', 'true');
     expect(child2).not.toHaveAttribute('tabindex');
   });
 
+  // UPDATED: This test now uses renderWithProviders to supply depth
   it('calculates the correct offset for parent node with icon', () => {
     const depth = 1;
-    const { container, getByText } = render(
-      <TreeNode
-        id="parent"
-        label="Parent Node"
-        depth={depth}
-        active="child1"
-        disabled={false}
-        selected={[]}
-        onTreeSelect={() => {}}
-        renderIcon={ArrowDown}>
-        <TreeNode id="child1" label="Child Node 1" disabled={false} />
-        <TreeNode id="child2" label="Child Node 2" disabled={true} />
-      </TreeNode>
+    const { getByText } = renderWithProviders(
+      <TreeNode id="parent" label="Parent Node" renderIcon={ArrowDown}>
+        <TreeNode id="child1" label="Child Node 1" />
+      </TreeNode>,
+      { depth }
     );
-    const treeNode = getByText('Parent Node');
-    expect(treeNode).toBeInTheDocument();
+
+    // FIX: Use .closest() to find the specific styled container
+    const styledNodeContainer = getByText('Parent Node').closest(
+      `.${prefix}--tree-node__label`
+    );
+    expect(styledNodeContainer).toBeInTheDocument();
```

Comment:
Looks like this was addressed?

```suggestion
    const styledNodeContainer = getByText('Parent Node').closest(
      `.${prefix}--tree-node__label`
    );
    expect(styledNodeContainer).toBeInTheDocument();
```

Reply:
Yeah this is addressed.